Makefile

```diff
@@ -71,11 +71,13 @@ OCCA_DIR ?= ../occa
 
 # env variable MAGMA_DIR can be used too
 MAGMA_DIR ?= ../magma
-# If CUDA_DIR is not set, check for nvcc, or resort to /usr/local/cuda
-CUDA_DIR  ?= $(or $(patsubst %/,%,$(dir $(patsubst %/,%,$(dir \
-               $(shell which nvcc 2> /dev/null))))),/usr/local/cuda)
+
+# Often /opt/cuda or /usr/local/cuda, but sometimes present on machines that don't support HIP
```

Comment:
HIP -> CUDA?

Reply:
Fixed, thanks.

As for the solution, this isn't the first time it came up, and I think some symmetry between CUDA and HIP autodetection is desirable. We could try `CUDA_HOME` and `CUDA_PATH` as alternatives for autodetection.